sdk/azidentity/device_code_credential.go

```diff
@@ -82,9 +83,9 @@ type DeviceCodeCredential struct {
 // options: Options used to configure the management of the requests sent to Azure Active Directory, please see DeviceCodeCredentialOptions for a description of each field.
 func NewDeviceCodeCredential(options *DeviceCodeCredentialOptions) (*DeviceCodeCredential, error) {
 	if options == nil {
-		temp := DefaultDeviceCodeCredentialOptions()
-		options = &temp
+		options = &DeviceCodeCredentialOptions{}
 	}
+	options.init()
```

Comment:
Perform `init()` on a copy.

Reply:
Same comment as AzureCLICredential. Let me know if the other solutions works!

Reply:
Done